java-squid/src/main/java/org/sonar/java/model/JavaVersionImpl.java

```diff
@@ -0,0 +1,100 @@
+/*
+ * SonarQube Java
+ * Copyright (C) 2012 SonarSource
+ * sonarqube@googlegroups.com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02
+ */
+package org.sonar.java.model;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.sonar.plugins.java.api.JavaVersion;
+
+public class JavaVersionImpl implements JavaVersion {
+
+  private static final Logger LOG = LoggerFactory.getLogger(JavaVersionImpl.class);
+
+  private static final int JAVA_7 = 7;
+  private static final int JAVA_8 = 8;
+  private final int javaVersion;
+  private final boolean initialized;
+
+  public JavaVersionImpl() {
+    this.javaVersion = -1;
+    this.initialized = false;
+  }
+
+  public JavaVersionImpl(int javaVersion) {
```

Comment:
No, as it is actually used by {{JavaCheckVerifier}} (allow testing behavior of rules with given version of java).